crates/client/src/client.rs

```diff
@@ -1080,6 +1102,32 @@ impl FuelClient {
         Ok(balance.amount)
     }
 
+    pub async fn balance_with_required_block_header(
+        &self,
+        owner: &Address,
+        asset_id: Option<&AssetId>,
+        required_block_height: u32,
+    ) -> io::Result<u128> {
```

Comment:
I'm a bit hesitant to have the `required_block_height` be a parameter here. It feels like we're cluttering the interface and are increasing the cognitive load for our users. This feels more like a state that should be maintained by the client and not even directly exposed to the user. 

I'd imagine the client maintaining a `chain_tip` parameter locally which is updated whenever it receives a response, and it's appended to the header of all requests made.

What are your thoughts on this?

Reply:
If I understood correctly you suggest to change the client to have the headers it will send in the requests in its state. I think that makes a lot of sense, let me have a go :)